### ISSUE TITLE
Allow validation without working credentials

### ISSUE THREAD

**Issue by thinkl33t** [pwaring/dns-sync-zone#41]
We'd like to validate passed zonefiles as part of our CI for DNS management, but given these branches are unprotected we'd risk leaking credentials.

I assume dry run needs to authenticate to allow --diffs to work, so if it isn't possible to add this to the existing dry run, could we add a --validate-only flag that exits cleanly after the validation phase.

**Comment by thinkl33t** [comment]
Currently we're working around this by checking for exit code 2, as this means validation has passed, and authentication has failed:

```
  python3 ./dns-sync-zone/sync-zone.py --dry-run --credentials-file "/tmp/credentials.json" --zone "${MBAPI_DOMAIN}" --rfc-file "${CI_PROJECT_DIR}/${MBAPI_DOMAIN}"
  # Exit code 2 is authentication failure, and means we've passed initial validation
  if [ $? -eq 2 ]; then
    exit 0
  else
    exit 1
  fi
```

**Comment by pwaring** [comment]
I think this shouldn't be too difficult - I can add a check to see if `--validate-only` has been passed, and if so the script will exit with status zero without bothering to instantiate the `mythic.MythicAPI` class (which triggers authentication).

**Comment by pwaring** [comment]
Change made and tested. Please can you check the `paul/validate-only` branch and see if it addresses the problem in your environment as well? If you pass the `--validate-only` option then the script should exit with a status of zero after the validation, or non-zero if the validation fails.

You'll still need to pass `--credentials-file`, as that's a required argument, although it doesn't matter what the contents are as the script will exit before trying to open it. That feels wrong to me so if the change I've made so far works for you I'll get a release pushed out and then investigate the `argparse` sub-parser functionality to see if I can make `--credentials-file` optional if `--validate-only` has been specified.

**Comment by thinkl33t** [comment]
This appears to work fine for our purposes, thanks Paul! :)

**Comment by pwaring** [comment]
Great, I'll get another release rolled out this evening.

**Comment by pwaring** [comment]
Released in v1.4.0.
